apps/dashboard/src/components/contract-components/contract-deploy-form/modular-contract-default-extensions-fieldset.tsx

```diff
@@ -56,45 +61,163 @@ export function ModularContractDefaultExtensionsFieldset(props: {
-                        }
-                      </FormErrorMessage>
-                    </FormControl>
-                  );
-                })}
-              </div>
-            </div>
+            <FormControl
+              key={formFieldKey}
+              isRequired
+              isInvalid={
+                !!form.getFieldState(formFieldKey, form.formState).error
+              }
+            >
+              <FormLabel> {param.name}</FormLabel>
+              <SolidityInput
+                solidityType={param.type}
+                solidityComponents={param.components}
+                variant="filled"
+                {...form.register(formFieldKey)}
+              />
+              <FormErrorMessage>
+                {
+                  form.getFieldState(formFieldKey, form.formState).error
+                    ?.message
+                }
+              </FormErrorMessage>
+            </FormControl>
           );
         })}
       </div>
     </div>
   );
 }
+
+function RenderPrimarySaleFieldset(prosp: {
+  extension: ExtensionWithIndex;
+  form: CustomContractDeploymentForm;
+  isTWPublisher: boolean;
+}) {
+  const { extension, form } = prosp;
+
+  const primarySaleRecipientPath =
+    `modularContractDefaultExtensionsInstallParams.${extension.extensionIndex}.primarySaleRecipient` as const;
+
+  return (
+    <PrimarySaleFieldset
+      isInvalid={
+        !!form.getFieldState(primarySaleRecipientPath, form.formState).error
+      }
+      register={form.register(primarySaleRecipientPath)}
+      errorMessage={
+        form.getFieldState(primarySaleRecipientPath, form.formState).error
+          ?.message
+      }
+    />
+  );
+}
+
+function RenderRoyaltyFieldset(props: {
+  extension: ExtensionWithIndex;
+  form: CustomContractDeploymentForm;
+  isTWPublisher: boolean;
+}) {
+  const { extension: ext, form } = props;
+
+  const royaltyRecipientPath =
+    `modularContractDefaultExtensionsInstallParams.${ext.extensionIndex}.royaltyRecipient` as const;
+
+  const royaltyBpsPath =
+    `modularContractDefaultExtensionsInstallParams.${ext.extensionIndex}.royaltyBps` as const;
+
+  return (
+    <RoyaltyFieldset
+      royaltyRecipient={{
+        isInvalid: !!form.getFieldState(royaltyRecipientPath, form.formState)
+          .error,
+        register: form.register(royaltyRecipientPath, {
+          required: "Required",
+        }),
+        errorMessage: form.getFieldState(royaltyRecipientPath, form.formState)
+          .error?.message,
+      }}
+      royaltyBps={{
+        isInvalid: !!form.getFieldState(royaltyBpsPath, form.formState).error,
+        value: form.watch(royaltyBpsPath) || "0",
+        setValue: (value) =>
+          form.setValue(royaltyBpsPath, value, {
+            shouldTouch: true,
+          }),
+        errorMessage: form.getFieldState(royaltyBpsPath, form.formState).error
+          ?.message,
+      }}
+    />
+  );
+}
+
+export function showRoyaltyFieldset(paramNames: string[]) {
```

Comment:
nit: do these actually need to be exported?

Reply:
yes - we are using in different page